pilot/pkg/model/service.go

```diff
@@ -1308,18 +1297,15 @@ func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
 func (s *Service) getAllAddressesForProxy(node *Proxy) []string {
 	if node.Metadata != nil && node.Metadata.ClusterID != "" {
 		addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
-		if (features.EnableDualStack || features.EnableAmbient) && len(addresses) > 0 {
+		if (features.EnableDualStack || features.EnableAmbient) && node.GetIPMode() == Dual {
```

Comment:
Looks like a good step towards detecting dual-stack and removing `ISTIO_DUAL_STACK` in the future.

Reply:
Maybe, but the true motivation is fixing if a pod is actually single stack(may reside in a remote cluster), we should not generate dual stack xds to it

Reply:
I am not sure we should filter them for SE, that is just an arbitrary lsit of IPs, could even be all IPv4?

Reply:
Why should distinguish k8s service and SE here? Any mismatched  ip protocol stack will not work

Reply:
ah I thought this was the only way to get multiple address. but sounds like L1305 can still

Reply:
Partially for my own understanding: 

Is this line even needed? `(features.EnableDualStack || features.EnableAmbient) && node.GetIPMode() == Dual` means `SupportsIPv4 == true` and `SupportsIPv6 == true`, so the `filterAddresses` on L1308 will always return the same result?

Just to be clear, I think the behavior is correct here just could be simplified (IIUC)

Reply:
It is a little confusing, filterAddresses only returns a kind of address. first returns ipv4